File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # complement = target - curr
        hashmap = {} # number -> index in nums
        for i, num in enumerate(nums):
            comp = target - num
            if comp in hashmap:
                return [i, hashmap[comp]]
            hashmap[num] = i
        

File: 0001-two-sum/0001-two-sum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
    hash = {}
    nums.each_with_index do |num, i|
        complement = target - num
        return [i, hash[complement]] if hash.key?(complement) && hash[complement] != i
        hash[num] = i
    end
end